tests_v2/unit/app/routes/datasets/test_download.py

```diff
@@ -160,3 +180,75 @@ async def httpx_response_coroutine(*args, **kwargs) -> httpx.Response:
         },
     )
     assert response.status_code == 200
+
+
+@pytest.mark.asyncio
+async def test_download_by_aoi_raster_csv(
+    generic_raster_version, apikey, async_client: AsyncClient
+):
+    dataset_name, version_name, _ = generic_raster_version
+    api_key, payload = apikey
+    origin = "https://" + payload["domains"][0]
+    headers = {"origin": origin, "x-api-key": api_key}
+
+    with (
+        patch(
+            "app.models.pydantic.datamart.AdminAreaOfInterest.get_geostore_id",
+            return_value=TEST_GEOSTORE_ID,
+        ),
+        patch(
+            "app.routes.datasets.downloads.get_geostore", return_value=TEST_GEOSTORE
+        ) as mock_get_geostore,
+        patch(
+            "app.routes.datasets.downloads._query_dataset_csv",
+            return_value=StringIO("x,y\n,1,2"),
+        ) as mock_query_dataset_csv,
+    ):
+        response = await async_client.get(
+            f"/dataset/{dataset_name}/{version_name}/download_by_aoi/csv?sql={TEST_SQL}&aoi[type]=admin&aoi[country]=IDN&aoi[region]=9&aoi[subregion]=9",
```

Comment:
Just a heads up, I'm going to advocate that we pass in `aio[provider]` and `aoi[version]` when we build these URLs in the Subscriptions MS. Obviously, the endpoints should still work. I just want to be explicit with the gadm source going forward.